### PR TITLE
proxy

### PR DESCRIPTION
**Pull request by tomazov**
advanced request settings

### CODE REVIEW

**Comment by tomazov** [comment]
advanced request settings

File: lib/Remote/HttpCommandExecutor.php [hunk 1]
```diff
@@ -182,10 +182,19 @@ class HttpCommandExecutor implements WebDriverCommandExecutor
...
         self::$w3cCompliantCommands = array_merge(self::$commands, self::$w3cCompliantCommands);
 
+        $uri = parse_url($http_proxy);
+        if ($uri !== false) {
+            $http_proxy = $uri['host'];
+            $http_proxy_port = $uri['port'];
+            $http_proxy_user = $uri['user'];
+            $http_proxy_pass = $uri['pass'];
+        }
+        unsewt($uri);
```
Comment (williamdes) [inline lib/Remote/HttpCommandExecutor.php:196]:
```suggestion
        unset($uri);
```

File: lib/Remote/HttpCommandExecutor.php [hunk 2]
```diff
@@ -182,10 +182,19 @@ class HttpCommandExecutor implements WebDriverCommandExecutor
      * @param string|null $http_proxy
      * @param int|null $http_proxy_port
```
Comment (williamdes) [inline lib/Remote/HttpCommandExecutor.php:183]:
```suggestion
     * @param int|null $http_proxy_port
     * @param string|null $http_proxy_user
     * @param string|null $http_proxy_pass
```

File: lib/Remote/HttpCommandExecutor.php (hunk 1 above)
Comment (OndraM) [inline lib/Remote/HttpCommandExecutor.php:196]: I would remove this whole part of parsing URI. If someone needs this (which would be very uncommon case), they can do this parsing themselves and provide the values directly.

**Review by OndraM — CHANGES_REQUESTED** [review]
While I can imagine there is possibility someone would run remote end behind a proxy, this PR is not even lintable... @tomazov Are you willing to finish the PR? If so, please:

1) Rebase the branch with current main branch
2) Address the comments in PR

Thanks!

**Pull request closed without merging**